code/datums/ai/basic_mobs/basic_ai_behaviors/basic_attacking.dm

```diff
@@ -16,6 +18,10 @@
 	set_movement_target(controller, target)
 
 /datum/ai_behavior/basic_melee_attack/perform(seconds_per_tick, datum/ai_controller/controller, target_key, targetting_datum_key, hiding_location_key)
+	if (cooldown_from_mob && isbasicmob(controller.pawn))
+		var/mob/living/basic/basic_pawn = controller.pawn
+		action_cooldown = basic_pawn.melee_attack_cooldown
```

Comment:
AI behaviors are singletons, I don't think it's correct to set the action cooldown for them

Reply:
To that end, consider Not Even affecting action cooldown, just make it abide by nextmove modifier

In the attack subtrees - if nextmove modifier > world.time return and don't finish planning rather than queuing a melee attack?